src/stores/ogp.tsx

```diff
@@ -0,0 +1,12 @@
+import { SWRResponse } from 'swr';
+
+import { restClient } from '~/utils/rest-client';
+import { useAuthenticationSWR } from '~/stores/use-authentication-swr';
+import { Ogp } from '~/domains/Ogp';
+
+export const useOgp = ({ url }: { url: string }): SWRResponse<Ogp, Error> => {
+  return useAuthenticationSWR(`ogp?url=${url}`, (endpoint: string) => restClient.apiGet(endpoint).then((result) => result.data), {
+    revalidateOnFocus: false,
+    revalidateOnReconnect: true,
+  });
+};
```

Comment:
url がない時はリクエストを飛ばさないべきなので


```suggestion
export const useOgp = (url?: string): SWRResponse<Ogp, Error> => {
  const key = url ? `ogp?url=${url}` : null
  return useAuthenticationSWR(key, (endpoint: string) => restClient.apiGet(endpoint).then((result) => result.data), {
    revalidateOnFocus: false,
    revalidateOnReconnect: true,
  });
};
```

Reply:
SWR は 第一引数が null だとリクエストを発行しない